spec/javascripts/govuk_publishing_components/load-analytics.spec.js

```diff
@@ -0,0 +1,323 @@
+  })
+
+  afterAll(function () {
+    window.GOVUK.analyticsGa4.vars = savedGa4Vars
+    window.GOVUK.analyticsVars = savedUaVars
+  })
+
+  describe('for GA4', function () {
+    beforeEach(function () {
+      window.GOVUK.analyticsGa4.vars = null
+      expected = {
+        id: 'GTM-MG7HG5W',
+        auth: undefined,
+        preview: undefined,
+        environment: 'production',
+        gem_version: 'not found'
+      }
+    })
+
+    it('loads GA4 on production domains', function () {
+      var domains = [
+        'www.gov.uk',
+        'www-origin.publishing.service.gov.uk'
+      ]
+
+      for (var i = 0; i < domains.length; i++) {
+        window.GOVUK.analyticsGa4.vars = null
+        var domain = domains[i]
+        window.GOVUK.loadAnalytics.loadGa4(domain)
+        expect(window.GOVUK.analyticsGa4.vars.environment).toEqual('production')
+      }
+    })
+
+    it('loads GA4 on staging domains', function () {
+      var domains = [
+        'www.staging.publishing.service.gov.uk',
+        'www.staging.publishing.service.gov.uk',
+        'www-origin.staging.publishing.service.gov.uk'
+      ]
+      for (var i = 0; i < domains.length; i++) {
+        window.GOVUK.analyticsGa4.vars = null
+        var domain = domains[i]
+        window.GOVUK.loadAnalytics.loadGa4(domain)
+        expect(window.GOVUK.analyticsGa4.vars.environment).toEqual('staging')
+      }
+    })
+
+    it('loads GA4 on integration domains', function () {
+      var domains = [
+        'www.integration.publishing.service.gov.uk',
+        'www.integration.publishing.service.gov.uk',
+        'www-origin.integration.publishing.service.gov.uk'
+      ]
+
+      for (var i = 0; i < domains.length; i++) {
+        window.GOVUK.analyticsGa4.vars = null
+        var domain = domains[i]
+        window.GOVUK.loadAnalytics.loadGa4(domain)
+        expect(window.GOVUK.analyticsGa4.vars.environment).toEqual('integration')
+      }
+    })
+
+    it('loads GA4 on development domains', function () {
+      var domains = [
+        'localhost', '127.0.0.1', '0.0.0.0', 'static.dev.gov.uk'
+      ]
+
+      for (var i = 0; i < domains.length; i++) {
+        window.GOVUK.analyticsGa4.vars = null
+        var domain = domains[i]
+        window.GOVUK.loadAnalytics.loadGa4(domain)
+        expect(window.GOVUK.analyticsGa4.vars.environment).toEqual('development')
+      }
+    })
+
+    it('ignores loading GA4 on unrecognised domains', function () {
+      var domains = [
+        'www.nationalarchives.gov.uk'
+      ]
+
+      for (var i = 0; i < domains.length; i++) {
+        window.GOVUK.analyticsGa4.vars = null
+        var domain = domains[i]
+        window.GOVUK.loadAnalytics.loadGa4(domain)
+        expect(window.GOVUK.analyticsGa4.vars).toEqual(null)
+      }
+    })
+
+    it('sets the correct production variables', function () {
+      window.GOVUK.loadAnalytics.loadGa4('www.gov.uk')
+      delete window.GOVUK.analyticsGa4.vars.internalDomains
```

Comment:
Why delete `internalDomains` here? Could you add `linkedDomains` to `expected`?

Reply:
@andysellick Depending on the order that the tests are run, either `internalDomains` exists or it doesn't - basically it depends on when the linkTracker tests run. As it isn't relevant to these tests I've deleted it.